src/pkg/reg/adapter/aliacr/openapi.go

```diff
@@ -0,0 +1,317 @@
+			break
+		}
+		page++
+	}
+	return tags, nil
+}
+
+// GetAuthorizationToken returns the authorization token for repository.
+func (acr *acrOpenapi) GetAuthorizationToken() (*authToken, error) {
+	var tokenRequest = cr.CreateGetAuthorizationTokenRequest()
+	var tokenResponse *cr.GetAuthorizationTokenResponse
+	tokenRequest.SetDomain(acr.domain)
+	tokenResponse, err := acr.client.GetAuthorizationToken(tokenRequest)
+	if err != nil {
+		return nil, err
+	}
+	var v authorizationToken
+	err = json.Unmarshal(tokenResponse.GetHttpContentBytes(), &v)
+	if err != nil {
+		return nil, err
+	}
+	return &authToken{
+		user:      v.Data.TempUserName,
+		password:  v.Data.AuthorizationToken,
+		expiresAt: v.Data.ExpireDate.ToTime(),
+	}, nil
+}
+
+type acreeOpenapi struct {
+	client     *cr_ee.Client
+	domain     string
+	instanceID string
+}
+
+var _ openapi = &acreeOpenapi{}
+
+// newAcreeOpenapi creates a new acreeOpenapi instance.
+func newAcreeOpenapi(accessKeyID string, accessKeySecret string, regionID string, instanceID string) (openapi, error) {
+	client, err := cr_ee.NewClientWithAccessKey(regionID, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, err
+	}
+	return &acreeOpenapi{
+		client:     client,
+		domain:     fmt.Sprintf(endpointTpl, regionID),
+		instanceID: instanceID,
+	}, nil
+}
+
+// ListNamespace returns a list of namespaces
+func (acree *acreeOpenapi) ListNamespace() ([]string, error) {
+	var namespaces []string
+	var nsReq = cr_ee.CreateListNamespaceRequest()
+	nsReq.SetDomain(acree.domain)
+	nsReq.InstanceId = acree.instanceID
+	var page = 1
+	for {
+		nsReq.PageNo = requests.NewInteger(page)
+		nsResp, err := acree.client.ListNamespace(nsReq)
+		if err != nil {
+			return nil, err
+		}
+		for _, ns := range nsResp.Namespaces {
+			namespaces = append(namespaces, ns.NamespaceName)
+		}
+		if !nsResp.ListNamespaceIsSuccess {
+			return nil, fmt.Errorf("failed to list namespace: %v", nsResp)
+		}
+		total, err := strconv.Atoi(nsResp.TotalCount)
+		if err != nil {
+			return nil, err
+		}
+		if total-(nsResp.PageNo*nsResp.PageSize) <= 0 {
+			break
+		}
+		page++
+	}
+	return namespaces, nil
+}
+
+// ListRepository returns a list of repositories in the specified namespace
+func (acree *acreeOpenapi) ListRepository(namespaceName string) ([]*repository, error) {
+	var repos []*repository
+	var reposReq = cr_ee.CreateListRepositoryRequest()
+	reposReq.SetDomain(acree.domain)
+	reposReq.InstanceId = acree.instanceID
+	reposReq.RepoNamespaceName = namespaceName
+	reposReq.RepoStatus = "NORMAL"
+	var page = 1
+	for {
+		reposReq.PageNo = requests.NewInteger(page)
+		reposResp, err := acree.client.ListRepository(reposReq)
+		if err != nil {
+			return nil, err
+		}
+		for _, repo := range reposResp.Repositories {
+			repos = append(repos, &repository{
+				Name:      repo.RepoName,
+				Namespace: repo.RepoNamespaceName,
+				ID:        repo.RepoId,
+			})
+		}
+		if !reposResp.ListRepositoryIsSuccess {
```

Comment:
Moved the check for success right after the response.

Reply:
done